src/edu/csus/ecs/pc2/core/report/FileComparisonUtilities.java

```diff
@@ -0,0 +1,500 @@
+                String valueTwo = Integer.toString(teamScoreRow.getRank());
+                FieldCompareRecord fieldCompareRecord = new FieldCompareRecord(fieldName, valueOne, valueTwo, null, valueOne);
+                fileComparison.addfieldCompareRecord(fieldCompareRecord);
+
+                // TODO 760 Do we need to compare problems??  firstRowfirstRow.getProblems() ?
+                fieldName = "num solved";
+                valueTwo = Long.toString(teamScoreRow.getScore().getNum_solved());
+                fieldCompareRecord = new FieldCompareRecord(fieldName, valueOne, valueTwo, null, valueOne);
+                fileComparison.addfieldCompareRecord(fieldCompareRecord);
+
+                fieldName = "total_time";
+                valueTwo = Long.toString(teamScoreRow.getScore().getTotal_time());
+                fieldCompareRecord = new FieldCompareRecord(fieldName, valueOne, valueTwo, null, valueOne);
+                fileComparison.addfieldCompareRecord(fieldCompareRecord);
+
+                fieldName = "team_id";
+                valueTwo = Long.toString(teamScoreRow.getTeam_id());
+                fieldCompareRecord = new FieldCompareRecord(fieldName, valueOne, valueTwo, null, valueOne);
+                fileComparison.addfieldCompareRecord(fieldCompareRecord);
+
+            }
+        } catch (Exception e) {
+            StaticLog.getLog().log(Level.WARNING, "Problem comparing scoreboard data/info", e);
+            e.printStackTrace(); // TODO 760 how to report problem to user ?
+        }
+
+        fileComparison.setNumberRows(numberRows);
+
+        return fileComparison;
+
+    }
+//   
+
+    public static FileComparison createAwardJSONFileComparison(String tsvFileName, String sourceDir, String targetDir, IFileComparisonKey fileComparisonKey) {
+
+        String firstFilename = sourceDir + File.separator + tsvFileName;
+        String secondFilename = targetDir + File.separator + tsvFileName;
+        FileComparison fileComparison = new FileComparison(firstFilename, secondFilename);
+
+        long numberRows = 0;
+        
+        try {
+            
+            List<CLICSAward> firstAwardRows = new ArrayList<CLICSAward>();
+            if (new File(firstFilename).exists()) {
+                firstAwardRows = loadAwardRows(fileComparison.getFirstFilename());
+            }
+            
+            List<CLICSAward> secondAwardRows =  new ArrayList<CLICSAward>();
+            if (new File(secondFilename).exists()) {
+                secondAwardRows = FileComparisonUtilities.loadAwardRows(fileComparison.getSecondFilename());
+            }
+            
+            /**
+             * Map first awards cisation to awards class
+             */
+            Map<String, CLICSAward> firstFileMap = new HashMap<String, CLICSAward>();
+            for (CLICSAward clicsAwardOne : firstAwardRows) {
+                String key = fileComparisonKey.getKey(clicsAwardOne);
+                firstFileMap.put(key, clicsAwardOne);
```

Comment:
This looks to me like the above `put()` could cause a failure under certain (legitimate) conditions.  A `CLICSAward` has a _citation_ field, and it is the (String) value of this _citation_ which is returned by the `filecomparison.getKey(clicsAwardOne)` call (one line earlier) - in other words, the _citation_ is used as the `key` in the `put()` call.  

It is legal, and totally logical, that more than one `CLICSAward` object could have the same _citation_ value -- for example, "Gold medal winner" (the example in the CLICS 22-07 spec, at https://ccs-specs.icpc.io/2022-07/contest_api#awards) -- there could be multiple "Gold medal winners" defined by separate `CLICSAward` instances.  If this happens, the second such award will wipe the first such award out of the hashmap (this is the definition of what happens when you `put()` something into a `Hashmap` using an existing key).    

As a result, the awards comparison (the `FileComparison` returned by this `createScoreboardJSONFileComparison()` method) will not contain an accurate comparison of the two award sets.

I consider this a SHOW STOPPER; I'd like either an explanation of why my logic here is wrong, or else a fix for this problem, before I would Approve this PR.

Reply:
I think that the conditions described would be very very unlikely.   Not impossible, just very very unlikely.
If you think this outlier should be addressed open a new issue and provide data that would reproduce the bug.
This will provide the team the opportunity to address this outlier rather than holding up this feature from being part of the next distribution.